docs/php/activities.md

```diff
@@ -3,3 +3,309 @@ id: activities
+    }
+
+    public function processFile(string $localName): string
+    {
+        // Implementation omitted for brevity.
+        return compressFile($this->localDirectory . $localName);
+    }
+
+    public function deleteLocalFile(string $fileName): void
+    {
+        unlink($this->localDirectory . $fileName);
+    }
+}
+```
+
+### Accessing Activity Info
+
+The [Activity](https://github.com/temporalio/sdk-php/blob/master/src/Activity.php) class provides static getters to access information about the Workflow that invoked it.
+Note that this information is stored in a thread local variable.
+Therefore, calls to Activity accessors succeed only in the process that invoked the Activity function.
+
+```php
+use Temporal\Activity;
+
+class FileProcessingActivitiesImpl implements FileProcessingActivities
+{
+    // ...
+
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->logger->info("namespace=" +  Activity.getNamespace());
+
+        $info = Activity::getInfo();
+
+        $this->logger->info("workflowId=" + $info->workflowExecution->getID());
+        $this->logger->info("runId=" + $info->workflowExecution->getRunID());
+        $this->logger->info("activityId=" + $info->id);
+        $this->logger->info("activityDeadline=" + $info->deadline);
+
+        // ...
+    }
+
+    // ...
+}
+```
+
+### Activity Heart Beating
+
+Some Activities are long-running.
+To react to a crash quickly, use a heartbeat mechanism.
+`Activity::heartbeat()` lets the Temporal service know that the Activity is still alive.
+You can piggyback `details` on an Activity heartbeat.
+If an Activity times out, the last value of `details` is included in the `TimeoutFailure` delivered to a Workflow.
+Then the Workflow can pass the details to the next Activity invocation.
+This acts as a periodic checkpoint mechanism for the progress of an Activity.
+
+```php
+use Temporal\Activity;
+
+class FileProcessingActivitiesImpl implements FileProcessingActivities
+{
+    // ...
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->dowloader->downloadWithProgress(
+            $bucketName,  
+            $remoteName,  
+            $localName,
+            // on progress
+            function ($progress) {
+                Activity::heartbeat($progress);
+            }
+        );
+
+        Activity::heartbeat(100); // download complete
+
+        // ...
+    }
+
+    // ...
+}
+```
+
+## Calling Activities
+
+`Workflow::newActivityStub` returns a client-side stub that implements an Activity interface.
+It takes Activity type and Activity options as arguments.
+Activity options are needed only if some of the required timeouts are not specified through the `#[ActivityMethod]` annotation.
+
+Calling (via `yield`) a method on this interface invokes an Activity that implements this method.
```

Comment:
This is not really correct. Calling a method on this interface returns a Promise without blocking for activity completion. Calling yield on that future blocks waiting for the activity completion. 

I think having two sections (as we do in the Java version of the docs) for sync and async doesn't make sense for PHP when the call always returns a Promise.

Reply:
I took a task to track this change.

@wolfy-j any chance we could sync on this section?